library/src/main/java/org/apache/apex/malhar/lib/dedup/BoundedDedupOperator.java

```diff
@@ -156,8 +156,7 @@ protected Slice getKey(Object tuple)
   protected void putManagedState(Object tuple)
   {
     Slice key = getKey(tuple);
-    ((ManagedTimeStateImpl)managedState).put(getBucketId(key), DEFAULT_CONSTANT_TIME,
-        key, new Slice(new byte[0]));
+    ((ManagedTimeStateImpl)managedState).put(getBucketId(key), DEFAULT_CONSTANT_TIME, key, new Slice(null, 0, 0));
```

Comment:
The test doesnot fail at this point. It fails for the test DeduperTimeBasedPOJOImplTest. 
This is because we try to create a BufferSlice (extends Slice) object in this code flow
`public BufferSlice(byte[] array, int offset, int length)`
`{`
`     super(array, offset, length);`
`}`
The above constructor call fails with IllegalArgumentException